Step2과제 수행중 full request는 되는데 리뷰요청이 안되는 문제를 해결하고자 테스트겸 full request를 요청합니다

anseonghyeon opened the pull request:
1. 디자인 토큰을 생성해서 emotion로 컴포넌트 디자인함
2. Mobile First Design을 위해 Layout 컴포넌트를 설정하여 max-width 를 720px로 제한함
3. 상단 네비게이션 바 영역을 구현함
4. 카테고리 섹션을 구현함
5. 기타 섹션을 구현함
6. 실시간 급상승 선물 랭킹을 구현함


아마 제가 step2 브랜치를 만들때 step1으로 만들어서 생긴 오류인것같습니다...
step2가 이미 많이 진행이 되어서 step2를 유지하고 해결하는 방법을 찾아보겠습니다...

anseonghyeon committed:
chore(repo): 홈과 로그인(WIP) 레포지토리 클론

프로젝트 초기화를 위해 홈과 로그인(WIP) 레포지토리를 클론함

anseonghyeon committed:
chore(repo): 불필요한 파일 및 코드 정리

프로젝트 내 사용하지 않는 파일과 코드를 삭제하여 구조를 정리함

anseonghyeon committed:
feat(config): 절대경로 import를 위한 alias 설정

vite와 typescript 설정 파일에 @ alias를 추가하여 src 하위 경로를 절대경로로 import할 수 있도록 함

anseonghyeon committed:
chore(config): Prettier 코드 포매터 추가

코드 스타일 유지를 위해 Prettier를 설정하고 관련 설정 파일을 추가함

anseonghyeon committed:
feat(style): emotion 스타일 라이브러리 추가 및 reset css 적용

emotion의 Global 스타일을 활용해 reset css를 전역에 적용함

anseonghyeon committed:
feat(style): 기본 폰트를 Pretendard로 적용

웹폰트 CDN과 글로벌 스타일에 Pretendard를 추가하여 전체 폰트를 일관되게 설정함

anseonghyeon committed:
feat(config): 불필요한 App.css index.css제거와 종속된 파일에서 import제거

anseonghyeon committed:
style(Main): emotion Color와 Typography를 설정하고 theme에서 Main의 background와 어울릴 만한 color를 선정해서 적용하였습니다

anseonghyeon committed:
feat(layout): Mobile First Design을 위해 Layout 컴포넌트를 만들어 max-width를 720px로 제한했습니다

anseonghyeon committed:
feat(layout): 네비게이션바를 구현하고 기타 섹션중 선물한 친구 섹션UI를 구현하였습니다.

anseonghyeon committed:
feat(layout): 카테고리 섹션을 구현하고 카테캠 화이팅 배너UI를 만들었습니다.

anseonghyeon committed:
feat(rank): 실시간 급상승 선물랭킹 섹션 UI를 구현했지만 아직 버튼 click되었을때 하이라이팅, 더보기 접기 기능이 구현되지않음 현재 full request 까진 동작하나 리뷰 요청이 안되는 문제가 있어 테스트차 먼저 push와 pull request를 해보겠습니다

anseonghyeon committed:
Merge branch 'anseonghyeon' into step2

anseonghyeon committed:
test: full request이후 수정사항을 push했을때 풀리퀘에도 적용되는지 테스트

anseonghyeon committed:
feat(ui): 버튼들의 하이라이팅 효과를 주었습니다

anseonghyeon commented:
리뷰 요청이 안되는 문제는 해결하였습니다

anseonghyeon committed:
feat(ui): 실시간 랭킹 접었다 펼수있는 더보기 접기 버튼 구현

Coreight98 commented on src/styles/GlobalStyle.tsx:
```diff
@@ -29,6 +29,7 @@ const reset = css`
   body {
     line-height: 1;
     font-family: 'Pretendard', -apple-system, BlinkMacSystemFont, system-ui, Roboto, 'Helvetica Neue', 'Segoe UI', 'Apple SD Gothic Neo', 'Noto Sans KR', 'Malgun Gothic', 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', sans-serif;
+    background-color: #f7f8fa;
```
body에 주입된 background color는 theme에 정의되어 있지 않은 컬러값인데, 이유가 궁금합니다!

이번 서비스에서 사용할 디자인적인 요소를 테마에 잘 정의해주셔서, 컬러, 폰트, 사이즈 와 같은 값들은 한곳에서 관리되면 추후 유지보수에도 용이할거 같아요!

anseonghyeon replied:
네, body의 background-color는 초반에 크롬 익스텐션인 Eye Dropper로 카카오톡 선물하기 예시 페이지의 body색을 찍어서 임의로 해둔것인데 디자인 토큰에 비슷한색이 있는지 확인하여 수정해두겠습니다!

anseonghyeon replied:
앗.. 리뷰요청은 실수로 2번눌렀습니다...

Coreight98 commented on src/styles/styled.d.ts:
```diff
@@ -0,0 +1,51 @@
+    };
+    spacing: Record<number, string>;
+    typography: {
+      [key: string]: {
```
테마에서 잘 정의해주셔서, 정의된 값들이 타입적으로 잘 추론되면 좋을거 같아요!

title1Bold 가 추론될 수 있도록 수정해도 좋을거 같습니다!

<img width="362" alt="스크린샷 2025-06-27 오후 3 07 19" src="https://github.com/user-attachments/assets/913af95a-4b22-4378-84ca-2c6389e15cae" />

추가적으로 저는 개인적으로 조금 더 잘게 쪼개서 정의를 하는 편입니다.
명확한 정답 보다는 시도해보시면 본인만의 기준을 만드시는것을 추천드립니다!

``` ts
type FontStyle = {
  fontSize:  string;
  fontWeight: number; // union type으로 정의해도 좋음
  lineHeight: string;
};

type FontType = 'Bold' | 'Regular';

typography: {
 body: {
  1: Record<FontType, FontStyle>
  }
  ...
}
```

Coreight98 commented on src/App.tsx:
```diff
@@ -1,35 +1,424 @@
 function App() {
-  const [count, setCount] = useState(0)
+  const [selected, setSelected] = useState('');
+  const [selected2, setSelected2] = useState('');
```
selected와 selected2는 각각 어떤 필터링을 수행하는 상태인지 판단하기 어려운 네이밍인거 같아요! 필터링의 역할을 보시고 바로 이해할 수 있는 네이밍을 선택해주시면 좋을거 같아요!
타입 추론이 될 수 있도록, 타입을 명시해주면 좋을거 같아요!

```suggestion
type RankType = 'ALL' | 'FEMALE' | 'MALE' | 'TEEN';
type SubRankType = 'WANT' | 'MANY' | 'WISH';
  const [rankType, setSelectedRankType] = useState<RankType|null>(null);
  const [subRankType, setSelectedSubRankType] = useState<SubRankType|null>(null);
```

anseonghyeon replied:
넵, 현재 step2 코드 리팩토링을 하고있습니다 버튼이 focus된 상태값을 저장하고 변경하는 변수, 함수들의 식별자들의 이름을 변경하였습니다.

anseonghyeon committed:
refactor(realtime-rank): 기존의 버튼이 포커싱된 상태를 표현하고 set하는 변수, 함수들의 식별자명을 변경하였습니다

바꾼이유는 기존의 selected1, selected2가 무슨 의미인지 한번에 와닫지 않았고 step3를 진행할때 코드작성시 방해가될것같아서 수정하였습니다

Coreight98 commented on src/App.tsx:
```diff
@@ -1,35 +1,424 @@
+            <RealtimeRankNav2Wrapper>
+              <RealtimeRankNav2Btn onClick={() => setSelected2('WANT')} isSelected={selected2 === 'WANT'}>받고 싶어한</RealtimeRankNav2Btn>
+              <RealtimeRankNav2Btn onClick={() => setSelected2('MANY')} isSelected={selected2 === 'MANY'}>많이 선물한</RealtimeRankNav2Btn>
+              <RealtimeRankNav2Btn onClick={() => setSelected2('WISH')} isSelected={selected2 === 'WISH'}>위시로 받은</RealtimeRankNav2Btn>
```
개선사항으로 공통적인 ui는 반복문을 통해 개선할 수 있을거 같아요!

예시
``` ts
const rankOptions = [
  { label: '받고 싶어한', value: 'WANT' },
  { label: '많이 선물한', value: 'MANY' },
  { label: '위시로 받은', value: 'WISH' },
] as const;

<RealtimeRankNav2Wrapper>
  {rankOptions.map(({ label, value }) => (
    <RealtimeRankNav2Btn
      key={value}
      onClick={() => setSelected2(value)}
      isSelected={selected2 === value}
    >
      {label}
    </RealtimeRankNav2Btn>
  ))}
</RealtimeRankNav2Wrapper>
```

anseonghyeon replied:
예시를 참고해서 코드 중복을 줄였습니다!

Coreight98 commented on src/App.tsx:
```diff
@@ -1,35 +1,424 @@
+              <RealtimeRankNavBtnTitleWrapper>
+                <RealtimeRankNavBtn onClick={() => setSelected('TEEN')} isSelected={selected === 'TEEN'}>👦🏻</RealtimeRankNavBtn>
+                <RealtimeRankNavTitle>청소년이</RealtimeRankNavTitle>
+              </RealtimeRankNavBtnTitleWrapper>
```
밑의 코멘트와 동일하게 반복되는 ui는 반복문을 활용해 개선할 수 있을거 같아요!

anseonghyeon replied:
넵, 수정해보겠습니다.

anseonghyeon replied:
map을 사용해서 코드중복을 줄였습니다

Coreight98 commented on src/App.tsx:
```diff
@@ -1,35 +1,424 @@
+  width: 100%;
+  height: auto;
+  margin-top: 20px;
+  display: grid;
```
그리드를 활용하셨군요! 👍😊

Coreight98 reviewed (changes requested):
성현님 안녕하세요! step2 과제 마무리 하시느라 고생 많으셨습니다!  🎉🎉

몇가지 코멘트 남겼는데, 확인 한번 부탁드립니다!

그리고 현재 메인 페이지 (app.tsx)에 모든 ui 가 들어있는데, 컴포넌트별 분리가 필요해보입니다!
저는 보통 코드를 작성할때, 다른 사람이 해당 코드를 보면 바로 이해할 수 있는가? 를 기준으로 작성하곤 해요! 한 파일에 코드가 길어지면 가독성을 해칠수도 있어서, 파일과 함께 컴포넌트레벨의 분리를 하곤 합니다! 과제 잘 해주셨는데, 해당 부분만 진행해주시면 더욱 완벽할거 같아요!


요 코멘트는 잘 해결해주셨고,https://github.com/next-step/react-gift-login/pull/79#discussion_r2170906588

추가적으로 아래 코멘트 들만 확인 부탁드릴게요
- https://github.com/next-step/react-gift-login/pull/79#discussion_r2170914346
- https://github.com/next-step/react-gift-login/pull/79#discussion_r2170915638

anseonghyeon committed:
refactor(rank): map을 사용해서 중복된 코드들을 줄였습니다

Coreight98 reviewed (approved):
고생많으셨습니다! 다음 과제는 로그인쪽 구현이네요! 화이팅입니다! 🥹🙌🏻
